src/git-source-provider.ts

```diff
@@ -92,12 +84,10 @@ export async function getSource(settings: ISourceSettings): Promise<void> {
       )
     }
 
-    // Remove possible previous extraheader
-    await removeGitConfig(git, authConfigKey)
-
+    const authHelper = gitAuthHelper.createAuthHelper(git, settings)
     try {
-      // Config extraheader
-      await configureAuthToken(git, settings.authToken)
+      // Configure auth
+      await authHelper.configureAuth()
```

Comment:
the auth helper `configureAuth` also handles removing possible previous extraheader